mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/impl/IdentifierConsumerDocumentProducer.java

```diff
@@ -214,7 +214,7 @@ private void indexAllQueue(SessionImplementor session, List<?> entities) throws
 		CompletableFuture.allOf( futures ).exceptionally( exception -> null ).join();
 
 		Integer completedFutures = this.countCompletedTasks( futures );
-		// TODO <<monitor>> indexingMonitor.documentAdded( completedFutures );
+		monitor.documentsAdded( completedFutures.longValue() );
```

Comment:
Just thought about this:

1. The "countCompletedTasks" method is useless at the moment. It counts all the tasks that are completed, either successfully or in error. Since we called `CompletableFuture.allOf( futures ).exceptionally( exception -> null ).join();` just two lines before, we know that all the futures completed in some way. So you could just do `futures.length` or `entities.size()`, and remove the `countCompletedTasks` method.
2. I originally though the "countCompletedTasks" method was counting the futures that succeeded. That could be a nice improvement. But if we did that, then we should add a `documentsFailed` method to the monitor to also inform it that some documents could not be indexed properly.

So... Let's just keep it simple for now, use `entities.size()`. And maybe create a ticket if you want to improve the monitor to differenciate successfully added documents from failed ones?

Reply:
Yes, I think we should create a follow-up ticket for that.